### PR TITLE
use latest build and BinaryProvider

### PR DESCRIPTION
**Pull request by visr**
The latest BinaryProvider allows updating already installed tarballs, and writes an updated `build.jl` such that I can use it as a dependency for `GDALBuilder`.
The build configuration itself did not change however. But still new hashes, a later BinaryBuilder was used to create them.

https://github.com/JuliaGeo/GEOSBuilder/compare/v3.6.2-0...v3.6.2-1

### CODE REVIEW

**Comment by yeesian** [comment]
LGTM

**Comment by visr** [comment]
Didn't realise I could just use the build script from the builder releases page... Using that now. Should be done now.

File: deps/build.jl [hunk 1]
```diff
@@ -1,39 +1,42 @@
...
-        error("Your platform $(Sys.MACHINE) is not supported by this package!")
+        install(url, tarball_hash; prefix=prefix, force=true, verbose=verbose)
     end
-
-    # Finally, write out a deps.jl file
-    write_deps_file(joinpath(@__DIR__, "deps.jl"), products)
+elseif unsatisfied
+    # If we don't have a BinaryProvider-compatible .tar.gz to download, complain.
+    # Alternatively, you could attempt to install from a separate provider,
+    # build from source or something even more ambitious here.
```
Comment (yeesian) [inline deps/build.jl]: should this comment go into the error message?

File: deps/build.jl [hunk 2]
```diff
@@ -1,39 +1,42 @@
-using BinaryProvider
+using BinaryProvider # requires BinaryProvider 0.3.0 or later
```
Comment (yeesian) [inline deps/build.jl]: this comment is already reflected in the REQUIRE file

File: deps/build.jl (hunk 1 above)
Comment (yeesian) [reply deps/build.jl]: see https://github.com/JuliaGeo/Proj4.jl/pull/18#issuecomment-385259109

File: deps/build.jl (hunk 2 above)
Comment (yeesian) [reply deps/build.jl]: https://github.com/JuliaGeo/Proj4.jl/pull/18#issuecomment-385259109

**Pull request merged**